lib/src/repository/check_repository.dart

```diff
@@ -2,29 +2,77 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'package:pana/src/pubspec.dart';
+import 'package:path/path.dart' as p;
+
+import '../logging.dart';
 import '../model.dart';
 import '../package_context.dart';
 
-import 'default_git_branch.dart';
+import 'git_local_repository.dart';
 import 'repository_url.dart';
 
 /// Returns the repository information for the current package.
 Future<Repository?> checkRepository(PackageContext context) async {
-  final sourceUrl = context.pubspec.repository ?? context.pubspec.homepage;
+  final sourceUrl = context.pubspec.repositoryOrHomepage;
   if (sourceUrl == null) {
     return null;
   }
   final url = RepositoryUrl.tryParse(sourceUrl);
   if (url == null) {
     return null;
   }
-  final detectedGitBranch = context.options.checkRemoteRepository
-      ? await tryDetectDefaultGitBranch(url.baseUrl)
-      : null;
-  final branch = detectedGitBranch ?? url.branch;
+  var branch = url.branch;
+  bool? isVerified;
+  String? verificationFailure;
+
+  void failVerification(String message, [error, StackTrace? st]) {
+    verificationFailure = message;
+    log.info(message, error, st);
+  }
+
+  if (context.options.checkRemoteRepository) {
+    GitLocalRepository? repo;
+    try {
+      repo = await GitLocalRepository.createLocalRepository(url.baseUrl);
```

Comment:
```suggestion
    final repo = await GitLocalRepository.createLocalRepository(url.baseUrl);
    try {
```

Reply:
it can't return a `repo` that should be deleted if it throws an exception -- so no need to enter try/finally before creating `repo`.

Reply:
Why does the declaration need to be outside the try-block?

Reply:
the finally block still applies...
anyway, this method is now only used in tests, so we may just remove it and do similar wrapping in tests...